sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -963,6 +963,7 @@ Scheduler::GraphBuildResult Scheduler::GraphBuilder::addCG(
       for (auto Ev = Events.begin(); Ev != Events.end();) {
         auto *EvDepCmd = static_cast<Command *>((*Ev)->getCommand());
         if (!EvDepCmd) {
+          ++Ev;
```

Comment:
Since this is the only change in the file, I'm approving the PR. However, in a broader scope, I don't really understand why after aborting fusion we keep processing events.

Reply:
What part do you refer to?

If you mean the `cancel_fusion` in the next `if` below, we are not cancelling fusion on the queue of the current command, but the fusion on another queue on which the current command has an event dependency.

Reply:
I think best solution is to add a comment documenting what @sommerlukas says. I'll follow this PR with a NFC with a comment there.

Reply:
Perhaps @victor-eds can add a clarifying comment in a separate patch and we merge this now to unblock the CI?

Reply:
Thank you both for clarifying. I do think a comment would help understanding it.